includes/class-scheduler.php

```diff
@@ -330,4 +338,86 @@ public static function schedule_profile_update( $user_id ) {
 			array( $user_id )
 		);
 	}
+
+	/**
+	 * Send a profile update when relevant user meta is updated.
+	 *
+	 * @param  int    $meta_id Meta ID being updated.
+	 * @param  int    $user_id User ID being updated.
+	 * @param  string $meta_key Meta key being updated.
+	 * @return void
+	 */
+	public static function user_update( $meta_id, $user_id, $meta_key ) {
+		// don't bother if the user can't publish
+		if ( ! \user_can( $user_id, 'publish_posts' ) ) {
+			return;
+		}
+		// the user meta fields that affect a profile.
+		$fields = array(
+			'activitypub_user_description',
+			'description',
+			'user_url',
+			'display_name',
+		);
+		if ( in_array( $meta_key, $fields, true ) ) {
+			self::schedule_profile_update( $user_id );
+		}
+	}
+
+	/**
+	 * Theme mods only have a dynamic filter so we fudge it like this.
+	 * @param  mixed $value
+	 * @return mixed
+	 */
+	public static function blog_user_update( $value = null ) {
+		self::schedule_profile_update( 0 );
+		return $value;
+	}
+
+	/**
+	 * Send a profile update to all followers. Gets hooked into all relevant options/meta etc.
+	 * @param int $user_id  The user ID to update (Could be 0 for Blog-User).
+	 */
+	public static function schedule_profile_update( $user_id ) {
+		\wp_schedule_single_event(
+			\time(),
+			'activitypub_send_update_profile_activity',
+			array( $user_id )
+		);
+	}
+
+	/**
+	 * Send an Actor Delete activity.
+	 * @param int $user_id  The user ID to Delete.
+	 */
+	public static function schedule_profile_delete( $user_id ) {
+		$user = get_userdata( $user_id );
+		if ( $user->has_cap( 'publish_posts' ) ) {
+			$temp_private_key = get_private_key_for( $user->ID );
+			add_option( 'activitypub_temp_sig_' . $user->ID, $temp_private_key );
+
+			$author_url = \get_author_posts_url( $user->ID );
+
+			$activity = new Activity();
+			$activity->set_id( $author_url . '#delete' );
+			$activity->set_type( 'Delete' );
+			$activity->set_actor( $author_url );
+			$activity->set_object( $author_url );
+			$activity->set_to( [ 'https://www.w3.org/ns/activitystreams#Public' ] );
```

Comment:
Total aside, but seeing the creation of an `Activity` object here, I think it might be nice to have a constructor shorthand, like:
```php
$activity = new Activity([
    'id' => $author_url . '#delete',
    'type' => 'Delete',
    'actor' => $author_url,
    'object' => $author_url,
    'to' => ['https://www.w3.org/ns/activitystreams#Public'],
]);
```

Just a little bit easier to mentally parse, for me anyway. Would need a `Base::__contstruct()` method added but the logic should be pretty simple. I'll maybe try it sometime. :)